processor/otel/consumer.go

```diff
@@ -58,3 +74,430 @@ func (c *Consumer) ConsumeTraceData(ctx context.Context, td consumerdata.TraceDa
 		Trace:          true,
 	})
 }
+
+func (c *Consumer) convert(td consumerdata.TraceData) (metadata.Metadata, []transform.Transformable) {
+	md := metadata.Metadata{}
+	parseMetadata(td, &md)
+	var hostname string
+	if md.System != nil && md.System.DetectedHostname != nil {
+		hostname = *md.System.DetectedHostname
+	}
+
+	transformables := make([]transform.Transformable, 0, len(td.Spans))
+	for _, span := range td.Spans {
+		if span == nil {
+			continue
+		}
+
+		var parentID *string
+		root := len(span.ParentSpanId) == 0
+		if !root {
+			str := fmt.Sprintf("%x", span.ParentSpanId)
+			parentID = &str
+		}
+		traceID := fmt.Sprintf("%x", span.TraceId)
+		spanID := fmt.Sprintf("%x", span.SpanId)
+		startTime := parseTimestamp(span.StartTime)
+		var duration float64
+		if span.EndTime != nil && !startTime.IsZero() {
+			duration = parseTimestamp(span.EndTime).Sub(startTime).Seconds() * 1000
+		}
+		name := span.GetName().GetValue()
+		if root || span.Kind == tracepb.Span_SERVER {
+			event := transaction.Event{
+				Id:        spanID,
+				ParentId:  parentID,
+				TraceId:   traceID,
+				Timestamp: startTime,
+				Duration:  duration,
+				Name:      &name,
+			}
+			parseTransaction(span, hostname, &event)
+			transformables = append(transformables, &event)
+		} else {
+			event := model_span.Event{
+				Id:        spanID,
+				ParentId:  *parentID,
+				TraceId:   traceID,
+				Timestamp: startTime,
+				Duration:  duration,
+				Name:      name,
+			}
+			parseSpan(span, &event)
+			transformables = append(transformables, &event)
+		}
+	}
+	return md, transformables
+}
+
+func parseMetadata(td consumerdata.TraceData, md *metadata.Metadata) {
+	serviceName := td.Node.GetServiceInfo().GetName()
+	if serviceName == "" {
+		serviceName = "unknown"
+	}
+	serviceName = truncate(serviceName)
+	md.Service = &metadata.Service{Name: &serviceName}
+
+	if ident := td.Node.GetIdentifier(); ident != nil {
+		if pid := ident.Pid; pid != 0 {
+			md.Process = &metadata.Process{Pid: int(pid)}
+		}
+		if hostname := truncate(ident.HostName); hostname != "" {
+			md.System = &metadata.System{DetectedHostname: &hostname}
+		}
+	}
+	if languageName, ok := languageName[td.Node.GetLibraryInfo().GetLanguage()]; ok {
+		md.Service.Language.Name = &languageName
+	}
+
+	switch td.SourceFormat {
+	case sourceFormatJaeger:
+		// version is of format `Jaeger-<agentlanguage>-<version>`, e.g. `Jaeger-Go-2.20.0`
+		versionParts := strings.SplitN(td.Node.GetLibraryInfo().GetExporterVersion(), "-", 3)
+		if v := versionParts[len(versionParts)-1]; v != "" {
+			md.Service.Agent.Version = &v
+		} else {
+			jaegerVersion := "unknown"
+			md.Service.Agent.Version = &jaegerVersion
+		}
+		agentName := agentNameJaeger
+		if md.Service.Language.Name != nil {
```

Comment:
The OTel Jaeger translator doesn't set `td.Node.Language`, so I think we should do that ourselves while parsing the version above. Then we can upstream the code. WDYT?

Reply:
SGTM, will add.